src/Monolog/Handler/DeduplicationHandler.php

```diff
@@ -70,13 +68,33 @@ public function flush(): void
             return;
         }
 
+        $gc = false;
+        $store = null;
+
+        if (file_exists($this->deduplicationStore)) {
+            $store = file($this->deduplicationStore, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
+            if (is_array($store)) {
+                $yesterday = time() - 86400;
+                for ($i = 0; $i < count($store); $i++) {
+                    if (substr($store[$i], 0, 10) < $yesterday) {
+                        $gc = true;
+                        break;
+                    }
+                }
```

Comment:
I would rather keep the gc property and have isDuplicate set it to true if an older entry is found, because the way this works here, if you override the file format in buildDeduplicationStoreEntry/isDuplicate, then you may break the GC functionality.

Reply:
Fair point! I moved the GC trigger back to `isDuplicate()`, readded the property and added a comment to `isDuplicate()` which hopefully explains to overwriters that they should set `$this->gc`.

I hope the failing CI workflow is not a huge problem. I am not sure what to do about it, because my changes do not seem to be responsible.

Reply:
Nope there's a mess on CI right now, I'm on it :)